kmip/pie/client.py

```diff
@@ -0,0 +1,380 @@
+        length_attribute = self.attribute_factory.create_attribute(
+            enums.AttributeType.CRYPTOGRAPHIC_LENGTH,
+            length)
+        mask_attribute = self.attribute_factory.create_attribute(
+            enums.AttributeType.CRYPTOGRAPHIC_USAGE_MASK,
+            [enums.CryptographicUsageMask.ENCRYPT,
+             enums.CryptographicUsageMask.DECRYPT])
+
+        # Create the template containing the attributes
+        attributes = [algorithm_attribute, length_attribute, mask_attribute]
+        template = cobjects.CommonTemplateAttribute(attributes=attributes)
+
+        # Create the asymmetric key pair and handle the results
+        result = self.proxy.create_key_pair(common_template_attribute=template)
+
+        status = result.result_status.enum
+        if status == enums.ResultStatus.SUCCESS:
+            public_uid = result.public_key_uuid.value
+            private_uid = result.private_key_uuid.value
+            return public_uid, private_uid
+        else:
+            reason = result.result_reason.enum
+            message = result.result_message.value
+            raise exceptions.KmipOperationFailure(status, reason, message)
+
+    @_synchronize
+    def register(self, secret):
+        """
+        Register a secret with a KMIP appliance.
+
+        Args:
+            secret (various): A secret to register. An instantiatable subclass
+                of ManagedObject from the Pie API.
+
+        Raises:
+            ClientConnectionNotOpen: if the client connection is unusable
+            KmipOperationFailure: if the operation result is a failure
+            TypeError: if the input argument is invalid
+        """
+        # Check input
+        if not isinstance(secret, pobjects.ManagedObject):
+            raise TypeError("secret must be a Pie ManagedObject")
+
+        # Verify that operations can be given at this time
+        if not self._is_open:
+            raise exceptions.ClientConnectionNotOpen()
+
+        # Extract and create attributes
+        attributes = list()
+        if hasattr(secret, 'cryptographic_usage_masks'):
+            mask_attribute = self.attribute_factory.create_attribute(
+                enums.AttributeType.CRYPTOGRAPHIC_USAGE_MASK,
+                secret.cryptographic_usage_masks)
+
+            attributes.append(mask_attribute)
+
+        template = cobjects.TemplateAttribute(attributes=attributes)
+        object_type = secret.object_type
+
+        # Register the secret and handle the results
+        secret = self.object_factory.convert(secret)
+        result = self.proxy.register(object_type, template, secret)
+
+        status = result.result_status.enum
+        if status == enums.ResultStatus.SUCCESS:
+            uid = result.uuid.value
+            return uid
+        else:
+            reason = result.result_reason.enum
+            message = result.result_message.value
+            raise exceptions.KmipOperationFailure(status, reason, message)
```

Comment:
The return values for different types of operations will be different, which would make the code in the `if` branch unique for each operation (e.g., see `create_key_pair` where we return both key IDs). Is it worth it to make a separate method for the `else` block?